### ISSUE TITLE
Requesting new Scan State option for DGPE

### ISSUE THREAD

**Issue by mummifiedclown** [hsoft/dupeguru#369]
Hi, please let me know if I'm off-base and there's actually some workaround, but let me see if I have this right:  When using the Reference state option for a scan folder, none of the contents of that folder (or any other folders marked as Reference) are compared against each other.  And when the results window is populated, those references that are matched against Normal scanned folders are hard set in that list as references and cannot be changed, even with the Re-Prioritize options.

What I would propose is either:
1. Having an option to turn off that hard set of the reference status after the scan is complete.
2. Better yet, keep the functionality of Reference and create a new scan state - not sure what to call it but AlreadyScanned is the gist of the idea - such that folders thus marked will not have any of their children compared against each other, but will be compared against children in other folders marked AlreadyScanned, Reference, or Normal.

The scenario is that when scanning a very large number of files, it can make sense to scan and process individual folders by themselves to weed out dups local to that folder structure, and then compare those folders against other processed folders without wasting time re-comparing their contents with themselves.  And at this stage it may not be feasible to declare an absolute reference so being able to Re-prioritize based on size or dimensions is key.  

In other words:
Scan FolderA as Normal and delete any duplicates within that folder.  Then exclude FolderA and scan FolderD as Normal, deleting duplicates within.  Then mark FolderA and FolderD as AlreadyScanned so that FolderA/FileB and FolderA/FileC won't be compared with each other, and FolderD/FileE and FolderD/FileF won't be compared with each other, but FileB and FileC will each be compared against FileE and FileF with none being tagged as an absolute reference.

Sorry about the wordy request - just want to make sure my logic is clear and sound...
Thanks much.

**Comment by hsoft** [comment]
I see what you mean. I'm not sure about the "AlreadyScanned" name, but I don't have a better idea.

Why don't you progressively add your folders to the scan?

1. Folder A as normal, clean dupes
2. Folder A as reference, Folder B as normal, clean dupes
3. Folder A and B as reference, Folder C as normal, clean dupes
4. And so on...

Wouldn't you achieve the same result?

**Comment by mummifiedclown** [comment]
Thanks for the response.  Only problem is that if subsequent folders added to the scans have versions that I want to keep, there doesn't appear to be any way to keep them if the previous folders are scanned using the Reference state - once a folder is scanned as Reference, any items from that folder that end up in the results can't be demoted.  Again, please correct me if I'm wrong and there's some workaround I'm missing.
